bundles/org.openhab.binding.bondhome/src/main/java/org/openhab/binding/bondhome/internal/handler/BondDeviceHandler.java

```diff
@@ -554,10 +554,12 @@ private synchronized void deleteExtraChannels(List<BondDeviceAction> currentActi
         Set<String> availableChannelIds = new HashSet<>();
 
         for (BondDeviceAction action : currentActions) {
-            String actionType = action.getChannelTypeId();
-            if (actionType != null) {
-                availableChannelIds.add(actionType);
-                logger.trace(" Action: {}, Relevant Channel Type Id: {}", action.getActionId(), actionType);
+            if (action != null) {
```

Comment:
Can `action` really be null ?

Reply:
There were null items in the currentActions list using the 3.4.0 release code:
![image](https://user-images.githubusercontent.com/26532542/209993886-f515439d-8c31-491e-9b2b-cb75a743cfdd.png)

Reply:
in case it is not a bug (I mean it is really expected to have some null entries), I would to replace in BondDevice  class:
```
    public List<BondDeviceAction> actions = Arrays.asList(BondDeviceAction.TURN_ON);
```
by
```
    public List<@Nullable BondDeviceAction> actions = Arrays.asList(BondDeviceAction.TURN_ON);
```

Reply:
I'd vote for making the list NonNullable, and fix whatever populates it to prune out actions that are not supported

Reply:
It is currently non-nullable and is set by the gson/json parser. I guess one could do some post processing in BondHttpApi.getDevice() after the gson parsing and before the BondDevice is returned... Something like device.actions.removeIf(null);

Reply:
Oh eww. So presumably gson is bypassing the NonNull restriction because there isn't an enum value for whatever your device's are advertising. Force removing nulls still sounds best to me, immediately after parsing. But I hope you have another PR coming to add support for your device's actions :)

Reply:
Fixed. The unsupported actions are "TiltClose" and "TiltOpen". Once I get time to dig into the device api, I will try to add the missing features.